src/components/OfferCard/OfferCard.js

```diff
@@ -0,0 +1,206 @@
+  padding: 10px 0;
+  ${({ withBorder, theme }) => withBorder
+    ? `border-bottom-width: 1px;
+       border-bottom-color: ${theme.colors.border};`
+    : ''}
+`;
+
+const CardInnerRow = styled.View`
+  flex-direction: row;
+  align-items: center;
+  justify-content: flex-end;
+  padding-left: 10px;
+  flex-wrap: wrap;
+`;
+
+const CardColumn = styled.View`
+  flex-direction: column;
+  align-items: ${props => props.alignRight ? 'flex-end' : 'flex-start'};
+  justify-content: flex-start;
+`;
+
+const CardText = styled(BaseText)`
+  ${fontStyles.regular};
+  letter-spacing: 0.18px;
+  color: ${({ label, theme }) => label ? theme.colors.text : theme.colors.secondaryText};
+  flex-wrap: wrap;
+  width: 100%;
+`;
+
+const CardButton = styled.TouchableOpacity`
+  flex-direction: row;
+  align-items: center;
+  padding: 4px 0;
+  margin-left: 10px;
+`;
+
+const ButtonLabel = styled(BaseText)`
+  color: ${({ secondary, theme }) => secondary ? theme.colors.secondaryText : theme.colors.primary};
+  font-size: ${fontSizes.regular}px;
+`;
+
+const ProviderIcon = styled(CachedImage)`
+  width: 24px;
+  height: 24px;
+`;
+
+const CardNote = styled(BaseText)`
+  flex-direction: row;
+  align-items: center;
+  padding: 4px 0;
+  margin-left: 10px;
+  color: ${themedColors.primary};
+  ${fontStyles.regular};
+`;
+
+const OfferCard = (props: Props) => {
+  const {
+    isDisabled,
+    onPress,
+    labelTop,
+    valueTop,
+    cardImageSource,
+    cardTopButton = {},
+    labelBottom,
+    valueBottom,
+    cardMainButton = {},
+    cardNote,
+  } = props;
+
+  const {
+    label: topButtonLabel,
+    onPress: topButtonOnPress,
+    isDisabled: isTopButtonDisabled,
+    isLoading: isTopButtonLoading,
+    isSecondary: isTopButtonSecondary,
+  } = cardTopButton;
+
+  const {
+    label: mainButtonLabel,
+    onPress: mainButtonOnPress,
+    isDisabled: isMainButtonDisabled,
+    isLoading: isMainButtonLoading,
+  } = cardMainButton;
+
+  return (
+    <ShadowedCard
+      contentWrapperStyle={{ paddingHorizontal: 16, paddingVertical: 6 }}
+      isAnimated
+      spacingAfterAnimation={10}
+      disabled={isDisabled}
+    >
+      <CardWrapper
+        disabled={isDisabled}
+        onPress={onPress}
+      >
+        <CardRow withBorder alignTop>
+          <CardColumn>
+            <CardText label>{labelTop}</CardText>
+            <CardText>{valueTop}</CardText>
+          </CardColumn>
+          <CardInnerRow style={{ flexShrink: 1 }}>
+            {!!cardImageSource && <ProviderIcon source={cardImageSource} resizeMode="contain" />}
+            {!!Object.keys(cardTopButton).length &&
+            <CardButton disabled={isTopButtonDisabled} onPress={topButtonOnPress}>
+              {!isTopButtonLoading
+                ? <ButtonLabel secondary={isTopButtonSecondary}>{topButtonLabel}</ButtonLabel>
+                : <Spinner width={20} height={20} />}
+            </CardButton>
+            }
```

Comment:
Yeah, I was thinking about similar approach - implement this loader in our current Button component. Since buttons might need some more facelifting, I skipped that for now.
But now when I think about it, it would not hurt that much :))
I'll make this adjustment and will push that in soon. Thanks :)